app/workers/subject_removal_worker.rb

```diff
@@ -0,0 +1,12 @@
+require 'subjects/remover'
+
+class SubjectRemovalWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :data_medium
+
+  def perform(subject_id)
+    Subjects::Remover.new(subject_id).cleanup
+  rescue Subjects::Remover::NonOrphan
```

Comment:
I know, but it feels dirty to have an exception be inside a common code path. Perhaps `#cleanup` could just return true/false depending on if it removed it?